### PR TITLE
[MM-50458] Desktop App release process

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
This contains a release process document for the Desktop App, outlining how to perform the release including the App Store work.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-50458

### CODE REVIEW

File: site/content/internal/desktop-release-process.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+5. Wait for the release candidate to finish building. You can monitor the progress [here](https://github.com/mattermost/desktop/actions/workflows/release.yaml). When it's finished, there will be a post in the `Release: Desktop Apps` channel with the new version number and a changelog.
+
+6. When the release process is finished, go to the `mattermost/desktop` GitHub repository and click on **Releases**.
+7. There should be a draft release of your new release candidate. Click the pencil to Edit.
+8. Make sure it's checked off as a pre-release, then click Publish Release.
+
+### Generating additional release candidates
+
+If there are any bugs reported by QA, once they are fixed we will need to generate an additional release candidate to verify that any issues has been fixed.
```
Comment (cwarnermm) [inline site/content/internal/desktop-release-process.md:65]:
```suggestion
If there are any bugs reported by QA, once they are fixed we will need to generate an additional release candidate to verify that any issues have been fixed.
```

**Review by cwarnermm — COMMENTED** [review]
Thanks, @devinbinnie! Provided editorial feedback inline

**Review by cwarnermm — APPROVED** [review]

File: site/content/internal/desktop-release-process.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+3. Copy the changelog for this version into the **"What's New in This Version"** section.
+4. In the Build section, select **Add Build**. Find the build corresponding to the version that was just built. 
+    - You should be able to recognize it by timestamp, but if not, grab the build from TestFlight and verify that the version number is correct.
+5. Make sure under Version Release that "Manually release this version" is selected.
+6. Once that's all done, you can select the **Add for Review** button and follow the prompts to submit a review.
+
+If the app is **approved** by App Review, you will get an email from App Store Connect saying "Your submission was accepted". From there, you can go back and release the app once the final release has been cut.
+
+If the app is **rejected** by App Review, you will get an email from App Store Connect saying "We noticed an issue with your submission". At that point, you'll need to log back into App Store Connect and review their comments. Make the necessary changes and you can follow the same process as above to re-submit for review until the app has been approved.
```
Comment (marianunez) [inline site/content/internal/desktop-release-process.md:100]: How long do they usually take for an answer?

**Review by marianunez — APPROVED** [review]
Thanks @devinbinnie! Instructions LGTM 👍 

This process looks like a good candidate for a playbook we can run in the Release: Desktop channel.

File: site/content/internal/desktop-release-process.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
+---
+title: Desktop Release Process
```
Comment (amyblais) [inline site/content/internal/desktop-release-process.md:2]: 1. If you'd like, feel free to add steps from this doc to the existing Playbook we have https://community.mattermost.com/playbooks/playbooks/h3a39biacpnuim7ufmwiuuoxfo/outline.
2. I can also add a link to this page from the Handbook release process page https://handbook.mattermost.com/operations/research-and-development/product/release-process/desktop-release.

**Review by harshilsharma63 — APPROVED** [review]

File: site/content/internal/desktop-release-process.md (hunk 2 above)
Comment (cwarnermm) [reply site/content/internal/desktop-release-process.md:111]: @devinbinnie - friendly bump on Maria's question.

File: site/content/internal/desktop-release-process.md (hunk 3 above)
Comment (cwarnermm) [reply site/content/internal/desktop-release-process.md:2]: @devinbinnie - friendly bump on Amy's feedback.

File: site/content/internal/desktop-release-process.md (hunk 2 above)
Comment (devinbinnie) [reply site/content/internal/desktop-release-process.md:111]: Sorry - missed this :D

Usually about 24 hours is their turnaround time, depends on the time of day. I'll add a note in here about that.

File: site/content/internal/desktop-release-process.md (hunk 3 above)
Comment (devinbinnie) [reply site/content/internal/desktop-release-process.md:2]: Missed this as well 😅 

I can have a look at the playbook and make sure these line up as well :)
A link to this page would definitely be helpful.

**Pull request merged**